main.js

```diff
@@ -2,9 +2,33 @@ import { courses } from './courses.js'
 
 const tbody = document.querySelector('tbody')
 const template = document.querySelector('#courserow')
+const filterBtn = document.querySelector('#filterConsent')
 
-courses.forEach(course => {
-  const row = template.content.cloneNode(true)
-  row.querySelector('td').textContent = `${course.crse} - ${course.descr}`
-  tbody.append(row)
+// 0 = Show all base case, 1 = consent needed, 2 = no consent needed
+let filterState = 0
```

Comment:
As far as I can see, everything related to filterState should be removed. The actual state of the checkbox should be used instead.